include/swift/AST/Expr.h

```diff
@@ -4213,23 +4213,22 @@ class DefaultArgumentExpr final : public Expr {
   unsigned ParamIndex;
 
   /// The source location of the argument list.
-  SourceLoc Loc;
+  SourceLoc ArgListLoc;
 
   /// Stores either a DeclContext or, upon type-checking, the caller-side
   /// default expression.
   PointerUnion<DeclContext *, Expr *> ContextOrCallerSideExpr;
 
 public:
   explicit DefaultArgumentExpr(ConcreteDeclRef defaultArgsOwner,
-                               unsigned paramIndex, SourceLoc loc, Type Ty,
-                               DeclContext *dc)
+                               unsigned paramIndex, SourceLoc argListLoc,
+                               Type Ty, DeclContext *dc)
     : Expr(ExprKind::DefaultArgument, /*Implicit=*/true, Ty),
-      DefaultArgsOwner(defaultArgsOwner), ParamIndex(paramIndex), Loc(loc),
-      ContextOrCallerSideExpr(dc) { }
+      DefaultArgsOwner(defaultArgsOwner), ParamIndex(paramIndex),
+      ArgListLoc(argListLoc), ContextOrCallerSideExpr(dc) { }
 
   SourceRange getSourceRange() const { return {}; }
```

Comment:
Seems like this shouldn't return an empty range if we know at least one location point. Does that break anything?

Reply:
I think that’s what we changed previously, it shouldn’t report a valid range since we don’t know where this argument is...

Reply:
I guess we'll find out shortly, but yeah this inconsistency doesn't happen anywhere else which is concerning. I'll add getSourceRange() back too and see if I can fix the verifier and tuple expression source range computation to account for / ignore the odd argument location it has, or maybe add a separate source location distinct from the location to use for #line/#column that doesn't trip them up.